index.js

```diff
@@ -237,11 +237,19 @@ RedisClient.prototype.create_stream = function () {
             this.stream.destroy();
         }
 
+        // Accept snake_case options and convert them to camelCase
+        var opts = {};
+        var keys = Object.keys(this.connection_options);
+        var key;
+        while (key = keys.pop()) {
+            opts[key.replace(/_([a-z])/g, function (g) { return g[1].toUpperCase(); })] = this.connection_options[key];
+        }
```

Comment:
The options are already cloned before they are passed to the constructor. This is done in the createClient function.